lib/Dialect/Calyx/Transforms/CalyxLoweringUtils.cpp

```diff
@@ -730,7 +737,8 @@ BuildBasicBlockRegs::partiallyLowerFuncToComp(mlir::func::FuncOp funcOp,
 
     for (auto arg : enumerate(block->getArguments())) {
       Type argType = arg.value().getType();
-      assert(isa<IntegerType>(argType) && "unsupported block argument type");
+      assert((isa<IntegerType>(argType) || isa<FloatType>(argType)) &&
```

Comment:
```suggestion
      assert(isa<IntegerType, FloatType>(argType) &&
```

(I think this should work)